packages/glimmer-demos/lib/ripples.ts

```diff
@@ -79,18 +79,14 @@ export function start() {
     data.points.push( new Point(e.timeStamp, e.offsetX, e.offsetY) );
   });
 
-  let baseOffset = Date.now() - window.performance.now();
```

Comment:
The original system works like this.

We will receive a timestamp on each mousemove event. We want to save that off. That's the time when the ripple spawned.

On every frame, we will get the current time and compute the current state of the ripple based on how much time has passed since the ripple has spawned.

We decided to get fancy and use performance.now for the second (on every frame, get the current time) part, but since p.now returns a number in ms since page load (navigation start), and the event timestamp is in ms since epoch, we need to convert between the two, hence the offset is added to the p.now to give the two number the same base.

However, in the latest stable version of Chrome, they made a breaking change to the web (:?!:) and changed the event timestamp to *also* use the navigation start time base. While it will indeed make our life easier here because we don't have to normalize the bases, other browsers of corse doesn't work that way. So if you test your branch in Firefox, it will not work :(

What they said is we can save a Date.now timestamp at page load, detect if the event timestamps are smaller or greater than that and decide what to do.










 ¯\_(ツ)_/¯